### DISCUSSION TITLE
Nodejs 12

### DISCUSSION THREAD

**Discussion by Vladislavik** [uNetworking/uWebSockets.js#133] [Q&A] [answered]
hi, on nodejs 12 it not work:
throw new Error('This version of µWS is not compatible with your Node.js build:\n\n' + e.toString());
		^
Error: This version of µWS is not compatible with your Node.js build:
Error: Cannot find module './uws_linux_x64_72.node'

**Comment by ghost** [comment, marked as answer]
Works now, latest

**Comment by ghost** [comment]
Node.js is really going downhill by every release.

In 10 I get 127k req/sec
In 12 I get 118k req/sec

And, you guessed it, in 11 I get 121k :face_with_thermometer:

**Comment by Vladislavik** [comment]
When will be new release to work with 12?

**Comment by waghanza** [comment]
@alexhultman Thanks for boosting performance of `node`, but `uWebSockets.js` seems **NOT** to be compatible with the latest `node` release (every 12.x)

**Comment by ghost** [comment]
I just ran it with Node.js 12 on Linux. The issue you referenced also lists one person confirming it works on macOS wiht Node.js 12 so I don't know. Pick any of the well known Linux:es

**Comment by yovanoc** [comment]
Yeah it's working for me on linux and mac perfectly with node 12.7

**Comment by waghanza** [comment]
@alexhultman @yovanoc it works

in fact, I was using using `15.10.0` instead of `15.11.0`

**Comment by colms** [comment]
FYI. I was getting the error `This version of µWS is not compatible with your Node.js build` when using Alpine but it worked with Debian, maybe because Alpine is musl and I'm using a pre-built binary which is probably built with libc.

**Comment by RJ4100** [comment]
@alexhultman I tried running on Node v14.16.0 and I confirmed the uWebsockets.js folder is in node_modules.  Error: Cannot find module 'uWebsockets.js' for node v14.16.0

**Reply by RJ4100** [reply]
@alexhultman Is there anything else that needs to be done on Ubuntu?

**Reply by RJ4100** [reply]
nvm on mac it works with uWebsockets.js linux need uWebSockets.js

**Comment by TheShubhendra** [comment]
Giving me the same error on Node Js v16.14.0

**Reply by e3dio** [reply]
Are you on Windows, see https://github.com/uNetworking/uWebSockets.js/issues/684

**Comment by alzalabany** [comment]
@alexhultman which version of nodejs do you recommend for production?

**Comment by Wyzix33** [comment]
As of v20.7.0 you can see this note: 

> 
> 
> Node.js 18, LTS-only
> 
> With this release and going forwards we will support only LTS versions of Node.js. Every year one new LTS version will be added and one old will be dropped. This means we always support a rolling set of 3 LTS versions. This release supports 14, 16 and 18.
